.eslintrc.js

```diff
@@ -62,48 +63,48 @@ module.exports = {
     //************************************************************************
     // Enforced
     //
+    curly: "error",
     // Set console calls to emit warnings as not enabled by eslint:recommended
     "no-console": "error",
     // Increase react-hooks/exhaustive-deps from warning to error
     "react-hooks/exhaustive-deps": "error",
-    // //************************************************************************
-    // // Relaxed: rules we are progressively adopting
-    // //
-    // // enforce types
-    // "@typescript-eslint/no-explicit-any": "warn",
-    // // enforce typed exports
-    // "@typescript-eslint/explicit-module-boundary-types": "warn",
-    // // enforce clean code
-    // "@typescript-eslint/no-unused-vars": "warn",
-    // // enforce interface naming convention
-    // "@typescript-eslint/naming-convention": [
-    //   "warn",
-    //   {
-    //     "selector": "interface",
-    //     "format": ["PascalCase"],
-    //     "custom": {
-    //       "regex": "^I[A-Z]",
-    //       "match": true
-    //     }
-    //   },
-    //   {
-    //     "selector": "typeAlias",
-    //     "format": ["PascalCase"],
-    //     "custom": {
-    //       "regex": "^T[A-Z]",
-    //       "match": true
-    //     }
-    //   }
-    // ],
-    // // enforce import ordering
-    // "sort-imports": [
-    //   "warn",
-    //   {
-    //     "allowSeparatedGroups": true,
-    //     "memberSyntaxSortOrder": ["none", "all", "single", "multiple"],
-    //     "ignoreCase": false
-    //   }
-    // ],
+    // enforce types
+    "@typescript-eslint/no-explicit-any": "error",
+    // Bans specific types from being used (e.g. builtin types)
+    "@typescript-eslint/ban-types": "error",
+    // enforce typed exports
+    "@typescript-eslint/explicit-module-boundary-types": "error",
+    // enforce clean code
+    "@typescript-eslint/no-unused-vars": "error",
```

Comment:
"@typescript-eslint/no-unused-vars": "error", I worked with this initially and reverted it back to warn. It can be quite disruptive during development/debugging as the compiler will block the serving of the dev server if it's flagged and we need to disable that rule each time that happens which is often as it's quite common to have unusued vars at that stage. If we make the build fail on warnings we can enforce it without making it stand in the way of development.

Reply:
When we have rules as warn, they are just ignored by the developers and we end up with too many build failures in the Pull Request (noise of PR not "ready", delays in terms of build infra consumption). Maybe we can work out something like having everything as warning by default in dev sandbox and enforce them at commit time with hooks, e.g. using `--max-warnings=0` , would that be good for you?

Reply:
Thanks, I think that would work - if we can enforce a warning-free code (early on with git hooks, and later in the pipeline), then all these rules will be enforced without disrupting the development workflow.

Reply:
@ahboujelben @alhugot-emr I think we can remove this now, can't we?

Thanks for the explanations and the handling of this, it looks really great like this!
We would even see default eslint warnings this way.